recovery/redis_users_recovery.py

```diff
@@ -102,18 +102,26 @@ def backup_data_from_redis(self):
             cursor, keys = self.redis.scan(cursor)
             for key in keys:
                 key_type = self.redis.type(key).decode()
-                if key_type == 'string':
-                    value = value if (value := self.redis.get(key)) is None else value.decode()
-                elif key_type == 'set':
-                    value = [member.decode() for member in self.redis.smembers(key)]
-                elif key_type == 'hash':
-                    hash_table = self.redis.hgetall(key)
-                    value = {hash_key.decode(): hash_table[hash_key].decode() for hash_key in hash_table}
-                else:
-                    exception = ValueError(
-                        f'Key of unknown type ({key_type}) was found while saving data from redis'
+                try:
+                    if key_type == 'string':
+                        value = value if (value := self.redis.get(key)) is None else value.decode()
+                    elif key_type == 'set':
+                        value = [member.decode() for member in self.redis.smembers(key)]
+                    elif key_type == 'hash':
+                        hash_table = self.redis.hgetall(key)
+                        value = {hash_key.decode(): hash_table[hash_key].decode() for hash_key in hash_table}
+                    else:
+                        raise ValueError(
+                            f'Key of unknown type ({key_type}) was found while saving data from redis'
+                        )
+                except ValueError as e:
+                    exception_message = str(e)
+                    self.logger.exception(exception_message)
```

Comment:
It's not necessary to pass the exception message as an argument here. `logger.exception()` fetches the message from the currently active exception, so it would appear in the log output twice.

Reply:
Fixed